lib/core/context.c

```diff
@@ -113,9 +113,13 @@ void h2o_context_init(h2o_context_t *ctx, h2o_loop_t *loop, h2o_globalconf_t *co
     memset(ctx->_module_configs, 0, sizeof(*ctx->_module_configs) * config->_num_config_slots);
 
     static pthread_mutex_t mutex = PTHREAD_MUTEX_INITIALIZER;
+    static int once = 0;
     pthread_mutex_lock(&mutex);
 
-    h2o_socketpool_register_loop(&ctx->globalconf->proxy.global_socketpool, loop);
+    if (once == 0) {
+        once++;
+        h2o_socketpool_register_loop(&ctx->globalconf->proxy.global_socketpool, loop);
+    }
```

Comment:
I think we do not need this change, as `h2o_socketpool_register_pool` has that check?

Reply:
It would make the code simpler, but i thought it wasn't great that each thread would overwrite `pool->_interval_cb.loop` with its own loop, making the result unpredictable. WDYT?

Reply:
>  i thought it wasn't great that each thread would overwrite pool->_interval_cb.loop with its own loop

I am not sure if that's the case.

The relevant code is https://github.com/h2o/h2o/blob/7fbc71a906786777b35ce6f4f571ca04886e3d66/lib/common/socketpool.c#L282-L285, which adopts the loop of the first invocation. I think it's ideally as using `once`.

Reply:
oh, of course. Thanks.